content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -0,0 +1,1286 @@
+
+Automatically inserted `apply_deferred` systems are optimized by automatically merging them if
+possible. In most cases, it is recommended to remove all manually inserted
+`apply_deferred` systems, as allowing Bevy to insert and merge these systems as needed will
+usually be both faster and involve less boilerplate.
+
+```rust
+// This will only add one apply_deferred system.
+app.add_systems(
+    Update,
+    (
+        (system_1_with_commands, system_2).chain(),
+        (system_3_with_commands, system_4).chain(),
+    )
+);
+```
+
+If this new behavior does not work for you, please consult the migration guide.
+There are several new APIs that allow you to opt-out.
+
+## One-Shot Systems Improvements
+
+<div class="release-feature-authors">authors: @Nathan-Fenner</div>
+
+In 0.12, we introduced [one-shot systems](https://bevyengine.org/news/bevy-0-12/#one-shot-systems), a handy way to call systems on demand without having to add them to a schedule.
+The initial implementation had some limitations with regards to what systems could and could not be used as one-shot systems.
+These limitations have since been resolved, starting with one-shot systems with input and output.
+
+```rust
+
+fn increment_sys(In(increment_by): In<i32>, mut counter: ResMut<Counter>) -> i32 {
+    counter.0 += increment_by;
+    counter.0
+}
+
+let mut world = World::new();
+let id = world.register_system(increment_sys);
+
+world.insert_resource(Counter(1));
+let count_one = world.run_system_with_input(id, 5).unwrap(); // increment counter by 5 and return 6
+let count_two = world.run_system_with_input(id, 2).unwrap(); // increment counter by 2 and return 8
+```
+
+Using either `world.run_system_with_input(system_id, input)` or `commands.run_system_with_input(system_id, input)`, you can now supply input parameters to systems that accept them. Additionally, both `world.run_system` and `world.run_system_with_input` now return system output as `Ok(output)`. Note that output cannot be returned when calling the system through commands, because of their deferred nature.
+
+Some smaller improvements to one-shot systems include registering boxed systems with `register_boxed_system` (which was shipped in 0.12.1, but didn't get a blog post) and the ability to register exclusive systems as one-shot systems.
+
+```rust
+world.register_system(|world: &mut World| { /* do anything */ });
+```
+
+All these improvements round out one-shot systems significantly: they should now work just like any other Bevy system.
+
+## WGPU Upgrade and Renderer Improvements
+
+<div class="release-feature-authors">authors: @Elabajaba</div>
+
+In Bevy 0.13 we upgraded from `wgpu` 0.17 to `wgpu` 0.19, which includes the long awaited `wgpu` [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to do things like compiling shaders asynchronously to avoid shader compilation stutters, and to multithread draw call creation for better performance.
```

Comment:
This should probably link https://github.com/bevyengine/bevy/pull/9172 in some way to show how we've taken advantage of this.

Reply:
Yeah, I wrote this up this way when @JMS55 was going to do a writeup for both async pipeline compilation and multithreaded render command encoding.